app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,20 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    available_roles = %i(master)
+    Minion.where(role: nil).each do |minion|
+      if available_roles.blank?
+        random_role = :minion
+      else
+        random_role = available_roles.delete_at(rand(available_roles.length))
+      end
+      minion.assign_role(role: random_role)
```

Comment:
For now this is ok, but this will take a lot of time. Would it be possible to add a websocket that informs the client what's going on ? For example, before you perform this call, send a "minion x will be deployed as a $role" message, and after that another message. Moreover, maybe it would also be worth it to send an initial message telling the client the amount of minions that are going to be assigned.

With all this, the client will be able to show more information to the user, so they are not clueless on what's going on 😄 

If you agree, maybe we can submit a new issue for this.